cpp/src/slice2py/PythonUtil.cpp

```diff
@@ -65,110 +67,94 @@ namespace
                 return string{builtinTable[builtin->kind()]};
             }
         }
-
-        ClassDeclPtr cl = dynamic_pointer_cast<ClassDecl>(type);
-        if (cl)
+        else if (auto cl = dynamic_pointer_cast<ClassDecl>(type))
```

Comment:
(minor) I would move these "misc contained types" (ClassDecl, Struct, Enum) to the last branch of the if. Something like:

```cpp
else
{
    auto contained = dynamic_pointer_cast<Contained>(type);
    assert(contained);
    return contained->mappedScoped(".");
}
```